lib/rules/valid-define-emits.js

```diff
@@ -77,11 +77,18 @@ module.exports = {
               ) {
                 if (
                   variable.defs.length &&
-                  variable.defs.every(
-                    (def) =>
-                      utils.inRange(scriptSetup.range, def.name) &&
-                      !utils.inRange(defineEmits.range, def.name)
-                  )
+                  variable.defs.every((def) => {
+                    if (def.type === 'ImportBinding') {
+                      return false
+                    }
+                    if (
+                      !utils.inRange(scriptSetup.range, def.name) ||
+                      utils.inRange(defineEmits.range, def.name)
+                    ) {
+                      return false
+                    }
+                    return true
+                  })
```

Comment:
Wouldn't this be much more readable and also much more concise?

```suggestion
                  variable.defs.every(
                    (def) =>
                      def.type !== 'ImportBinding' &&
                      utils.inRange(scriptSetup.range, def.name) &&
                      !utils.inRange(defineEmits.range, def.name)
                  )
```